sql/src/test/java/io/crate/integrationtests/SysShardMinLuceneVersionTest.java

```diff
@@ -76,4 +71,36 @@ public void testMinLuceneVersion() throws Exception {
                "test_upgrade_required_parted| STARTED| 5.5.2| 5\n" +
                "test_upgrade_required_parted| UNASSIGNED| NULL| 10\n"));
     }
+
+    @Test
+    public void testUpgradeSegments() throws Exception {
+        startUpNodeWithDataDir("/indices/cluster_checks/cratedata_lucene_min_version.zip");
+        execute("select table_name, routing_state, min_lucene_version, count(*) from sys.shards " +
+                "where table_name IN " +
+                "('test_upgrade_required', 'test_upgrade_required_parted', 'test_blob_upgrade_required') " +
+                "group by table_name, routing_state, min_lucene_version order by 1, 2, 3");
```

Comment:
test would make more sense if query filters just old segments and upgrades them (no old segments after upgrading at all)